### PR TITLE
[Cassandra] Add metric_type mapping for the fields of  `metrics` datastream.

### PR DESCRIPTION
**Pull request by ritalwar**
- Enhancement

## What does this PR do?

This PR adds metric_type mapping for the fields of `metrics` datastream.
## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates #6708

### CODE REVIEW

File: packages/cassandra/data_stream/metrics/fields/fields.yml [hunk 1]
```diff
@@ -32,59 +39,80 @@
...
       fields:
         - name: completed
           description: compaction completed tasks.
           type: long
+          metric_type: gauge
         - name: pending
           description: compaction pending tasks.
           type: long
+          metric_type: gauge
     - name: dropped_message
```
Comment (kush-elastic) [inline packages/cassandra/data_stream/metrics/fields/fields.yml:116]: If we are not sure that these fields are counter then can we map it as a gauge?

Comment (ritalwar) [reply packages/cassandra/data_stream/metrics/fields/fields.yml:116]: These metrics fall into either the Timer or Meter category, so didn't map it to gauge. 
https://cassandra.apache.org/doc/latest/cassandra/operating/metrics.html#dropped-metrics

@lalit-satapathy thoughts?

Comment (lalit-satapathy) [reply packages/cassandra/data_stream/metrics/fields/fields.yml:116]: Which metric here is of doubt?

@kush-elastic any pending reviews?

Comment (ritalwar) [reply packages/cassandra/data_stream/metrics/fields/fields.yml:116]: Raised an [issue](https://github.com/elastic/integrations/issues/6967) to take care of these fields when there is support for metric_type `timer` and `meter` .

Comment (richtalwar) [reply packages/cassandra/data_stream/metrics/fields/fields.yml:116]: After conducting a thorough analysis, I have assigned the metric_type "counter" to these fields. This decision was made based on the fact that there are multiple attributes related to dropped messages, and our pipeline specifically processes the "Count" attribute for all types of dropped messages. It is important to note that the "Count" attribute itself is of type counter.

For further clarification and better understanding, please refer to the [link](https://github.com/ritalwar/integrations/blob/7e0edb35d44925af04917d3db09a065054cf100d/packages/cassandra/data_stream/metrics/agent/stream/stream.yml.hbs#L248) and screenshots.
image
image

Comment (ritalwar) [reply packages/cassandra/data_stream/metrics/fields/fields.yml:116]: After conducting a thorough analysis, I have assigned the metric_type "counter" to these fields. This decision was made based on the fact that there are multiple attributes related to [dropped](https://cassandra.apache.org/doc/latest/cassandra/operating/metrics.html#dropped-metrics) messages, and our pipeline specifically processes the "Count" attribute for all types of dropped messages. It is important to note that the "Count" attribute itself is of type counter.

For further clarification and better understanding, please refer to the [link](https://github.com/ritalwar/integrations/blob/7e0edb35d44925af04917d3db09a065054cf100d/packages/cassandra/data_stream/metrics/agent/stream/stream.yml.hbs#L248) and screenshots.

Screenshot 2023-07-19 at 12 50 44 PM

Screenshot 2023-07-19 at 12 51 13 PM

**Review by kush-elastic — APPROVED** [review]

File: packages/cassandra/data_stream/metrics/fields/fields.yml (hunk 1 above)
Comment (lalit-satapathy) [reply packages/cassandra/data_stream/metrics/fields/fields.yml:116]: @agithomas, Can you confirm if this mapping is correct?

**Review by agithomas — APPROVED** [review]
LGTM, as Richa as validated with Cassandra documentation

**Pull request merged**